### PR TITLE
adds support for alerts and triggers on group by based sigma rules 

### PR DESCRIPTION
**Pull request by eirsep**
### Description
- For each group by rule a bucket level monitor is created.
- Bucket level monitor doesn't support the current detector triggers which cater focussedly to per document alerting 
- This change uses workflow's chained monitor findings feature to feed findings generated by all bucket level monitors as source data to a match all doc level monitor. The chained findings monitor simply creates findings for all docs which match the bucket level monitor rule and honors the detector trigger condition creating alerts if there are any group by rules mentioned in the trigger.

Behaviour of non-group by rule matching is unaffected.

### CODE REVIEW

File: src/test/java/org/opensearch/securityanalytics/resthandler/DetectorMonitorRestApiIT.java [hunk 1]
```diff
@@ -940,7 +941,7 @@ else if (ruleId == minRuleId) {
         // Assert findings
         assertNotNull(getFindingsBody);
         // 8 findings from doc level rules, and 3 findings for aggregation (sum, max and min)
-        assertEquals(11, getFindingsBody.get("total_findings"));
+        assertEquals(19, getFindingsBody.get("total_findings"));
```
Comment (sbcd90) [inline src/test/java/org/opensearch/securityanalytics/resthandler/DetectorMonitorRestApiIT.java:944]: should we hide the findings generated by match of `_id: *` queries?

File: src/test/java/org/opensearch/securityanalytics/resthandler/DetectorMonitorRestApiIT.java [hunk 2]
```diff
@@ -1409,7 +1410,7 @@ public void testCreateDetector_verifyWorkflowExecutionBucketLevelDocLevelMonitor
         Map<String, Object> getFindingsBody = entityAsMap(getFindingsResponse);
 
         assertNotNull(getFindingsBody);
-        assertEquals(6, getFindingsBody.get("total_findings"));
+        assertEquals(10, getFindingsBody.get("total_findings"));
```
Comment (sbcd90) [inline src/test/java/org/opensearch/securityanalytics/resthandler/DetectorMonitorRestApiIT.java:1413]: same comment as https://github.com/opensearch-project/security-analytics/pull/545/files#r1319008403

Comment (eirsep) [reply src/test/java/org/opensearch/securityanalytics/resthandler/DetectorMonitorRestApiIT.java:1413]: let's go with this for now
will do de-dupe later.

File: src/test/java/org/opensearch/securityanalytics/resthandler/DetectorMonitorRestApiIT.java (hunk 1 above)
Comment (eirsep) [reply src/test/java/org/opensearch/securityanalytics/resthandler/DetectorMonitorRestApiIT.java:944]: been thinking about this

let's go with this for now
will do de-dupe later.

File: src/main/java/org/opensearch/securityanalytics/transport/TransportIndexDetectorAction.java [hunk 3]
```diff
@@ -785,7 +850,7 @@ private IndexMonitorRequest createBucketLevelMonitorRequest(
          triggers.add(bucketLevelTrigger1);
          } **/
 
-        Monitor monitor = new Monitor(monitorId, Monitor.NO_VERSION, detector.getName(), false, detector.getSchedule(), detector.getLastUpdateTime(), null,
+        Monitor monitor = new Monitor(monitorId, Monitor.NO_VERSION, detector.getName() + UUID.randomUUID(), false, detector.getSchedule(), detector.getLastUpdateTime(), null,
```
Comment (sbcd90) [inline src/main/java/org/opensearch/securityanalytics/transport/TransportIndexDetectorAction.java:853]: is this change needed? i reverted this & integ tests still work fine.

**Review by sbcd90 — APPROVED** [review]

**Comment by eirsep** [comment]
https://github.com/opensearch-project/security-analytics/issues/558

**Review by getsaurabh02 — APPROVED** [review]
